src/ucp/tag/tag_send.c

```diff
@@ -291,36 +332,16 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_tag_send_sync_nb,
         goto out;
     }
 
-    req = ucp_request_get(ep->worker);
-    if (req == NULL) {
-        ret = UCS_STATUS_PTR(UCS_ERR_NO_MEMORY);
+    req = ucp_tag_match_get_request(ep->worker, param);
```

Comment:
how does FORCE_IMM_CMPL work for this function? seems wrong to just ignore it

Reply:
FORCE_IMM_CMPL is ignored here.
we can add extra branch to check this flag or add docs about this flag is ignored